src/D2L.CodeStyle.Analyzers/Immutability/MutabilityInspector.cs

```diff
@@ -283,10 +283,19 @@ HashSet<ITypeSymbol> typeStack
 			ExpressionSyntax expr,
 			HashSet<ITypeSymbol> typeStack
 		) {
-			if( expr.Kind() == SyntaxKind.NullLiteralExpression ) {
+
+			switch( expr.Kind() ) {
 				// This is perhaps a bit suspicious, because fields and
 				// properties have to be readonly, but it is safe...
-				return MutabilityInspectionResult.NotMutable();
+				case SyntaxKind.NullLiteralExpression:
+
+				// Lambda initializers for readonly members are safe
+				// because they can only close over other members, which
+				// will be checked independently, or static members of
+				// another class, which are also analyzed
+				case SyntaxKind.SimpleLambdaExpression:
+				case SyntaxKind.ParenthesizedLambdaExpression:
+					return MutabilityInspectionResult.NotMutable();

```

Comment:
Did we decide this was safe regardless of the type it was assigned to, and implicit conversions to another type would get us into a potentially-mutable state?

Reply:
We've got another issue open to deal with implicit conversions (#35). It's definitely a problem, but a pretty general one.